### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by Trishana9**
- [x] Product listing
- [ ] Pagination
- [ ] Product detail
- [ ] Slider in product detail

### CODE REVIEW

File: src/components/ProductListing.vue [hunk 1]
```diff
@@ -17,6 +17,48 @@
...
+      }
+  },
+  async mounted()
+  {
+      let res=await axios.get('https://dummyjson.com/products');
+      console.log(res)
+      this.productList=res.data.products
+      console.log(this.productList)
+  }
+};
```
Comment (lamasagar) [inline src/components/ProductListing.vue]: make the api calls in created hook,

**Review by lamasagar — CHANGES_REQUESTED** [review]

File: src/components/ProductListing.vue (hunk 1 above)
Comment (Trishana9) [reply src/components/ProductListing.vue]: added api calls and created hook

File: src/components/ProductListing.vue [hunk 2]
```diff
@@ -1,22 +1,54 @@
...
+  data()
+  {
+      return{
+          productList:[],
+          current_page: 1,
+          product_per_page: 20,
+      }
+  },
+  methods:{
+      calculateDiscoutedAmount(product){
```
Comment (lamasagar) [inline src/components/ProductListing.vue:41]: you can destructure in the argument like
```
discountAmount({price, discountPercentage}) {
   return (price-((price*discountPercentage)/100)).toFixed(2)
}
```

File: src/components/ProductListing.vue [hunk 3]
```diff
@@ -1,22 +1,54 @@
...
+          return (product.price-((product.price*product.discountPercentage)/100)).toFixed(2)
+      }
+
+  },
+  created()
+  {
+      axios.get('https://dummyjson.com/products')
+      .then((response) => this.productList = response.data.products)
```
Comment (lamasagar) [inline src/components/ProductListing.vue:49]: same here,

`.then(({data}) => this.productList = data.products)`

**Review by lamasagar — APPROVED** [review]
try adding routing

**Pull request merged**